crates/rspack_plugin_javascript/src/visitors/swc_visitor/provide.rs

```diff
@@ -0,0 +1,168 @@
+use rspack_core::Provide;
+use swc_core::common::DUMMY_SP;
+use swc_core::common::{Span, SyntaxContext};
+use swc_core::ecma::ast::{
+  CallExpr, Callee, ComputedPropName, Expr, ExprOrSpread, Ident, ImportDecl, Lit, MemberExpr,
+  MemberProp, ModuleDecl, ModuleItem, Str,
+};
+use swc_core::ecma::visit::{as_folder, Fold, VisitMut, VisitMutWith};
+
+pub fn provide_builtin<'a>(opts: &'a Provide) -> impl Fold + 'a {
+  as_folder(ProvideBuiltin::new(opts))
+}
+
+pub struct ProvideBuiltin<'a> {
+  opts: &'a Provide,
+}
+
+impl<'a> ProvideBuiltin<'a> {
+  pub fn new(opts: &'a Provide) -> Self {
+    ProvideBuiltin { opts }
+  }
+
+  fn handle_ident(&self, ident: &mut Ident) -> Expr {
+    if let Some(module_path) = self.opts.get(&ident.sym.to_string()) {
+      // println!("Before Ident transformation Span: {:?}", ident.span);
+      let new_ident = self.create_obj_expr(ident.span, module_path);
+      // println!("After Ident transformation Span: {:?}", new_ident);
+      new_ident
+    } else {
+      Expr::Ident(ident.clone())
+    }
+  }
+
+  fn handle_member_expr(&self, member_expr: &mut MemberExpr) -> Expr {
+    let identifier_name = self.get_nested_identifier_name(member_expr);
+    if let Some(module_path) = self.opts.get(&identifier_name) {
+      // println!("Before Member transformation Span: {:?}", member_expr.span);
+      let new_expr = self.create_obj_expr(member_expr.span, module_path);
+      // println!("After Member transformation Span: {:?}", new_expr);
+      new_expr
+    } else {
+      Expr::Member(member_expr.clone())
+    }
+  }
+
+  fn create_obj_expr(&self, span: Span, module_path: &[String]) -> Expr {
+    let call_expr = self.create_call_expr(
+      Span {
+        lo: span.lo,
+        hi: span.hi,
+        ctxt: SyntaxContext::from_u32(2),
```

Comment:
You could get the unresolved `SyntaxContext` by using `SyntaxContext::empty().apply_mark(unresolved_mark)`. 

You could get the unresolved `Mark`  here https://github.com/web-infra-dev/rspack/blob/93975ed98266be67475a0d14bf9501d89cd9134d/crates/rspack_plugin_javascript/src/visitors/mod.rs#L68